cloudinit/net/renderers.py

```diff
@@ -38,13 +43,13 @@ def search(priority=None, target=None, first=False):
         if render_mod.available(target):
             cur = (name, render_mod.Renderer)
             if first:
-                return cur
+                return [cur]
```

Comment:
This is much easier to reason about if we're always returning a list. The only place this function is being called is from the `select` function below (which I modified accordingly) and a couple unit tests.